src/formatter.rs

```diff
@@ -211,6 +213,18 @@ impl<W: Write> ColorFormatter<W> {
             } else if !is_bright(&new_style.fg_color) && is_bright(&self.current_style.fg_color) {
                 queue!(self.output, SetAttribute(Attribute::Reset))?;
             }
+            if new_style.bold != self.current_style.bold {
+                if new_style.bold.unwrap_or_default() {
+                    queue!(self.output, SetAttribute(Attribute::Bold))?;
+                } else {
+                    // NoBold results in double underlining on some terminals, so we use reset
+                    // instead. However, that resets other attributes as well, so we reset
+                    // our record of the current style so we re-apply the other attributes
+                    // below.
+                    queue!(self.output, SetAttribute(Attribute::Reset))?;
+                    self.current_style = Style::default();
```

Comment:
I don't know which is better, but I think it's also fine to always reset and apply merged attributes and colors.

Reply:
I'm also not sure, but I've made some other changes to make us emit fewer escapes. I'll close this, but let me know if you think I should change anything.